creusot/src/backend/term.rs

```diff
@@ -123,9 +123,13 @@ impl<'tcx, N: Namer<'tcx>> Lower<'_, 'tcx, N> {
 
                     Exp::qvar(of_qname).app([Exp::qvar(to_qname).app([self.lower_term(arg)])])
                 }
+                TyKind::RawPtr(ty1, _) if let TyKind::RawPtr(ty2, _) = term.ty.kind() && ty1 == ty2 => {
+                    self.lower_term(arg)
```

Comment:
```suggestion
                    // Note: coercing between different pointer types is more complicated because of fat pointers metadata.
                    self.lower_term(arg).coerce(term.ty)
```